docs/coordinate_systems.md

```diff
@@ -0,0 +1,177 @@
+
+### Grayskull harvesting
+
+On Grayskull there is no harvesting. That means that on each Grayskull chip full grid of tensix cores (12x10) is available.
+
+Harvesting of non-tensix cores (DRAM, PCIe, ARC, Ethernet) is also not supported.
+
+### Wormhole harvesting
+
+On wormhole, harvesting of tensix rows is supported. That means that on the tensix grid (8x10) there will always be 8 columns of chips, but number of rows can decrease. In practice, Wormhole chips have one or two rows harvested. Example for two harvested rows is in the image below.
+
+![Wormole harvesting](images/tensix_harvested_rows.png)
+
+Note that there is no limitation on which specific rows can be harvested.
+
+Harvesting of non-tensix cores (DRAM, PCIe, ARC, Ethernet) is not supported on Wormhole.
+
+### Blackhole harvesting
+
+On Blackhole, harvesting of tensix columns is supported. That means that on the tensix grid (14x10), there will always be 10 rows of tensix cores, but number of columns may decrease. In practice, Blackhole chips have (TODO: how many columns) columns harvested. Example for two harvested columns is in the image below.
+
+(TODO: attach image of harvested columns)
+
+Note that there is no limitation on which specific columns can be harvested.
+
+(TODO: dram harvesting)
+
+Harvesting of other cores (PCIe, ARC, Ethernet) is not supported on Blackhole.
+
+# Coordinate systems
+
+In the following section, coordinate systems that can be used to access cores are going to be described. For each coordinate system it will be described how does harvesting affect access to different cores when using certain coordinate system, as well as the API support through UMD for this coordinate system.
+
+To keep everything consistent, example of 8x10 grid, which represents Wormhole chip, is going to be used in each image in this section.
+
+In order to illustrate harvesting effect on coordinate systems, example with two harvested rows of Tensix cores (image below) is going to be used
+
+## Logical coordinates
+
+This coordinate system is mostly used to reference Tensix cores, since this cores are most frequently accessed. This coordinate system hides the details of physical coordinates and allows upper layers of the stack to access Tensix endpoints through a set of traditional Cartesian Coordinates. This coordinate systems has very simple indexing, it starts from `0-0` and ends at `(X-1)-(Y-1)` where X and Y is number of cores on x-axis and y-axis, respectively. Example on logical coordinate indexing is one the image below
+
+![Tensix logical coordinates](images/tensix_logical_coordinates.png)
+
+Logical coordinates are used in a same way for Grayskull, Wormhole and Blackhole, only difference being in grid size of tensix cores.
```

Comment:
So if you are looking down at the chip, logical coordinates are the system where you can simply count cores either horizontally (X) or vertically (Y), _skipping rows that have been harvested_?

Reply:
So if you harvest 8x10 (note this is XxY annotation) grid with 2 rows (y coordinate), you get 8x8 grid. You can index both x and y in range [0-7], and you don't care which tensix cores are harvested, you will just see that the grid of useful tensix cores has shrinked to 8x8

Reply:
Logical coordinates are for someone who doesn't care about perf that much, for example relative position of Tensix cores to DRAM cores, but you still want to run something on the chip. It is a nice representation of the standard way people are used to view the grid [0-length)

Reply:
Or is logical where the the corner (0,0) a Tensix, or whatever is actually in the corner?

Reply:
Based on your example, logical coordinates are Tensix only, i.e. an ERISC or PCIe or DRAM core does not exist?
Or may have a negative coordinate?